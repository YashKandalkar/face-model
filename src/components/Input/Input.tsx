import React from "react";

interface Props {
  className?: string;
  placeholder?: string;
  type?: string;
  name?: string;
  autoComplete?: boolean;
  autoCapitalize?: boolean;
  spellCheck?: boolean;
}

const Input = (
  {
    className,
    placeholder,
    type,
    name,
    autoComplete,
    autoCapitalize,
    spellCheck,
  }: Props,
  ref: React.Ref<HTMLInputElement>
) => {
  const [isOccupied, setOccupied] = React.useState(false);

  const changeTheme = (event: React.FormEvent<HTMLInputElement>) => {
    if (event.currentTarget.value === "") {
      setOccupied(false);
    } else {
      setOccupied(true);
    }
  };

  return (
    <React.Fragment>
      <input
        className={className}
        placeholder={placeholder}
        type={type}
        name={name}
        autoComplete={autoComplete ? "on" : "off"}
        autoCapitalize={autoCapitalize ? "words" : "off"}
        spellCheck={spellCheck ? "true" : "false"}
        ref={ref}
        onChange={changeTheme}
      />
      <style jsx>{`
        input {
          background-color: ${isOccupied ? "#484848" : "#2b2b2b"};
          border-radius: 0.25rem;
          caret-color: #fff;
          color: #f5f5f5;
          display: block;
          font-family: inherit;
          font-size: inherit;
          font-weight: inherit;
          line-height: 50px;
          outline: none;
          padding: 0 14px;
          transition: all 250ms cubic-bezier(0.4, 0, 0.2, 1) 0s;
          width: 100%;
        }
        input::placeholder {
          color: #a9a9a9;
        }
      `}</style>
    </React.Fragment>
  );
};

export default React.forwardRef(Input);
