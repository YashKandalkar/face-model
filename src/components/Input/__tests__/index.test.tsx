import React from "react";
import { mount } from "enzyme";
import Input from "../index";

describe("Input", () => {
  it("should be defined", () => {
    expect(Input).toBeDefined();
  });

  it("should be rendered", () => {
    const wrapper = mount(<Input />);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should forward ref by default", () => {
    const ref = React.createRef<HTMLInputElement>();
    const wrapper = mount(<Input ref={ref} />);
    expect(ref.current).not.toBeNull();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should display placeholder", () => {
    const ref = React.createRef<HTMLInputElement>();
    const wrapper = mount(<Input placeholder="text" ref={ref} />);
    expect(ref.current?.placeholder).toBe("text");
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should get current value", () => {
    const ref = React.createRef<HTMLInputElement>();
    const wrapper = mount(<Input ref={ref} />);
    expect(ref.current?.value).toBe("");
    wrapper.find("input").getDOMNode<HTMLInputElement>().value = "text";
    wrapper.find("input").simulate("change");
    expect(ref.current?.value).toBe("text");
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should match snapshot", () => {
    expect(<Input placeholder="Input" />).toMatchSnapshot();
  });
});
