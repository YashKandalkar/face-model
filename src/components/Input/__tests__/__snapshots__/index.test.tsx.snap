// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<ForwardRef(Input)
  placeholder="Input"
/>
`;
